wire/README.md

```diff
@@ -491,3 +491,116 @@ Create a new struct that includes the app plus all of the dependencies you want
 to mock. Create a test-only injector that returns this struct, give it providers
 for the concrete mock types, and use `wire.Bind` to tell Wire that the
 concrete mock types should be used to fulfill the appropriate interface.
+
+## Frequently Asked Questions
+
+### How does Wire relate to other Go dependency injection tools?
+
+Other dependency injection tools for Go like [dig][] or [facebookgo/inject][]
+are based on reflection. Wire run as a code generator, which means that the
+injector works without making calls to a runtime library. This enables easier
+introspection of initialization and correct cross-references for tooling like
+[guru][].
+
+[dig]: https://github.com/uber-go/dig
+[facebookgo/inject]: https://github.com/facebookgo/inject
+[guru]: https://golang.org/s/using-guru
+
+### How does Wire relate to other non-Go dependency injection tools (like Dagger 2)?
+
+Wire's approach was inspired by [Dagger 2][]. However, it is not the aim of Wire
+to emulate dependency injection tools from other languages: the design space and
+requirements are quite different. For example, the Go compiler does not support
+anything like Java's annotation processing mechanisms. The difference in
+languages and their idioms necessarily requires different approaches in
+primitives and API.
+
+[Dagger 2]: https://google.github.io/dagger/
+
+### Why is Wire part of Go Cloud?
+
+Wire was designed to reduce the toil in setting up the multitude of dependencies
+often found when interacting with common cloud providers. However, Wire is not
+coupled to Go Cloud or cloud servers: Wire is a standalone tool.
+
+The Go Cloud team is investigating splitting Wire into its own repository.
+Follow [#513][] for updates.
+
+[#513]: https://github.com/google/go-cloud/issues/513
+
+### Why use pseudo-functions to create provider sets or injectors?
+
+In the early prototypes, Wire directives were specially formatted comments. This
+seemed appealing at first glance as this meant no compile-time or runtime
+impact. However, this unstructured approach becomes opaque to other tooling not
+written for Wire. Consider tools like [`gorename`][] or [guru][]: it would not
```

Comment:
Is there a link to the blog post yet? Consider adding one somewhere in the README....
https://blog.golang.org/wire

Reply:
Added one at the top.